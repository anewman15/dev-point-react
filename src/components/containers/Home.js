/* eslint-disable camelcase */
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

const Home = ({ currentUser }) => {
  const { username, first_name, profile_image_url } = currentUser;
  return (
    <div className="my-3 mx-auto px-2 md:px-8 h-screen filter drop-shadow-2xl">
      <div className="mx-auto md:w-11/12 lg:w-10/12 my-6 p-2 rounded border border-yellow-50 h-full bg-gray-50">
        <div className="p-4 flex justify-center space-x-8 items-center content-center">
          <div className="w-32 rounded-full">
            <img className="rounded-full" src={profile_image_url || `${process.env.PUBLIC_URL}/dev.png`} alt="profile_image" />
          </div>
          <p className="text-3xl text-primary-800 text-center font-bold">
            Welcome
            {first_name ? ` ${first_name} !` : ` ${username} !`}
          </p>
        </div>

        <hr className="mx-8 border-1 border-gray-200" />

        <div className="my-8 px-4 mx-auto w-8/12 text-gray-800">
          <p className="py-3 leading-loose">
            To add a profile image and other details, please visit the Edit Profile page
            <Link to="/edit_profile" className="btn btn-warning ml-4">Edit&nbsp;Profile</Link>
          </p>
          <p className="py-3">
            To get into business, navigate to our
            <Link className="px-2 link-primary" to="/developers">Developers </Link>
            page and get in touch with them
          </p>
        </div>
      </div>
    </div>
  );
};

Home.propTypes = {
  currentUser: PropTypes.object,
}.isRequired;

const mapStateToProps = state => ({
  currentUser: state.currentUser,
});

export default connect(mapStateToProps)(Home);
