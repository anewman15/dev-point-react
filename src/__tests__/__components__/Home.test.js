import { unmountComponentAtNode } from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from '../../components/containers/Home';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Home', () => {
  it('renders the Home component to the dom', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Home />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
