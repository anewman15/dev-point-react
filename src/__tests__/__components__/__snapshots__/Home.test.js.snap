// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders the Home component to the dom 1`] = `
<div
  className="my-6 p-3"
>
  <p
    className="my-3 px-2 is-size-1 has-text-centered has-text-weight-bold"
  >
    WTH!
  </p>
  <p
    className="my-3 px-2 is-size-3 has-text-centered"
  >
    This place is supposed to be your Home
  </p>
  <p
    className="my-3 px-2 is-size-5 has-text-centered"
  >
    But, looks like there is nothing here!
  </p>
  <div
    className="my-5 is-flex is-justify-content-center"
  >
    <a
      className="button is-warning has-text-success has-text-center"
      href="https://www.furniturebari.com/"
      rel="noreferrer"
      target="_blank"
    >
      Go Buy Some Furniture
      <span
        className="px-2 has-text-weight-bold"
      >
        NOW
      </span>
    </a>
  </div>
  <div
    className="my-6 px-3"
  >
    <p
      className="has-text-center py-3"
    >
      Pun aside, this page will soon allow you to add a profile picture, company info and an about me/us section
    </p>
    <div
      className="py-3 is-flex is-justify-content-center is-align-items-center"
    >
      <p
        className="has-text-center"
      >
        For now though, navigate to our
        <a
          className="px-2 has-text-primary has-text-weight-bold"
          href="/developers"
          onClick={[Function]}
        >
          Developers 
        </a>
        page and get in touch with them
      </p>
    </div>
  </div>
</div>
`;
